web-server/src/constants/log-formatter.ts

```diff
@@ -0,0 +1,31 @@
+export interface ParsedLog {
+  timestamp: string;
+  logLevel: string;
+  message: string;
+  role?: string;
+  ip?: string;
+}
+
+export const generalLogRegex =
+  /^\[(.*?)\] \[(\d+)\] \[(INFO|ERROR|WARN|DEBUG|WARNING|CRITICAL)\] (.+)$/;
+export const httpLogRegex =
+  /^(\S+) (\S+) (\S+) \[([^\]]+)\] "([^"]*)" (\d+) (\d+) "([^"]*)" "([^"]*)"$/;
+export const redisLogRegex =
+  /^(?<role>\d+:[XCMS]) (?<timestamp>\d{2} \w{3} \d{4} \d{2}:\d{2}:\d{2}\.\d{3}) (?<loglevel>[\.\-\#\*]) (?<message>.*)$/;
+export const postgresLogRegex =
+  /^(\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d{3} \w+) \[(\d+)\] (\w+):(.+)(?:\n(?:\s+.*)?)*$/m;
+export const postgresMultiLineLogRegex =
+  /^(?<timestamp>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d{3} UTC) \[\d+\] (?<loglevel>[A-Z]+):\s*(?<message>(.|\n)+?)$/;
+export const dataSyncLogRegex = /\[(\w+)\]\s(.+?)\sfor\s(\w+)\s(.+)/;
+export const validLogLevels = new Set([
+  'DEBUG',
+  'INFO',
+  'NOTICE',
+  'WARNING',
+  'ERROR',
+  'LOG',
+  'FATAL',
+  'PANIC',
+  'STATEMENT',
+  'DETAIL'
+]);
```

Comment:
this should be an enum in our `resources` file

Reply:
I added these as enums and also removed the validLogLevels logic from the parser. Previously, I was using two approaches to parse PostgreSQL logs, but during testing, I noticed that my first approach, PostgresMultilineLogMatch, can parse both types of PostgreSQL logs. Therefore, there's no need to write separate logic for each type ,refer to the [official postgres documentation  ](https://www.postgresql.org/docs/current/runtime-config-logging.html#:~:text=server%20command%20line.-,log_line_prefix,-(string)). and  [this article ](https://betterstack.com/community/guides/logging/how-to-start-logging-with-postgresql/#structured-logging-in-json:~:text=%25m%20%5B%25p%5D%20%25q,the%20following%20details%3A) also for  [log levels ](https://www.postgresql.org/docs/current/runtime-config-logging.html#:~:text=Table%C2%A019.2.-,Message%20Severity%20Levels,-Severity).